### PR TITLE
[CONTINT-3679] Add workqueue telemetry

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds telemetry to the SBOM scan queue.

### Motivation
We would like a better observability on the kubernetes scan queue.

### Describe how to test/QA your changes
Deploy the agent with sbom collection and telemetry enabled, configure an openmetrics check configured on it. Make sure the telemetry is collected: https://dddev.datadoghq.com/notebook/7744286/workqueue-telemetry

On helm:
```
agents:
  podAnnotations:
    ad.datadoghq.com/agent.checks: |-
      {
        "openmetrics": {
          "init_config": {},
          "instances": [
            {
              "openmetrics_endpoint": "http://localhost:6000/telemetry",
              "namespace": "datadog.agent",
              "metrics": [".*"]
            }
          ]
        }
      }
datadog:
    env:
    - name: DD_TELEMETRY_ENABLED
      value: "true"
    sbom:
    host:
      enabled: true # I have tested with false as there is currently a bug on main
    containerImage:
      enabled: true
      uncompressedLayersSupport: true
```

### CODE REVIEW

File: pkg/sbom/telemetry/telemetry.go [hunk 1]
```diff
@@ -82,3 +85,120 @@ var (
...
+	g.Gauge.Set(v)
+}
+
+type counterWrapper struct {
+	telemetry.Counter
+}
+
+// Inc implements the workqueue.CounterMetric interface
+func (c counterWrapper) Inc() {
+	c.Counter.Inc()
```
Comment (AliDatadog) [inline pkg/sbom/telemetry/telemetry.go:117]: The workqueue accepts only `Inc()` and our counter implements `Int(tags ...string)`

File: pkg/sbom/scanner/scanner.go [hunk 2]
```diff
@@ -52,11 +52,15 @@ type Scanner struct {
...
 func NewScanner(cfg config.Config, wmeta optional.Option[workloadmeta.Component]) *Scanner {
 	return &Scanner{
-		scanQueue: workqueue.NewRateLimitingQueue(
+		scanQueue: workqueue.NewRateLimitingQueueWithConfig(
 			workqueue.NewItemExponentialFailureRateLimiter(
 				cfg.GetDuration("sbom.scan_queue.base_backoff"),
 				cfg.GetDuration("sbom.scan_queue.max_backoff"),
 			),
+			workqueue.RateLimitingQueueConfig{
+				Name:            "sbom",
```
Comment (clamoriniere) [inline pkg/sbom/scanner/scanner.go:61]: `sbom_request` maybe to be more accurate and not misunderstanding with the SBOM report.

File: pkg/sbom/telemetry/telemetry.go [hunk 3]
```diff
@@ -82,3 +85,125 @@ var (
...
+		"SBOM queue adds",
+		commonOpts,
+	)}
+}
+
+// NewLatencyMetric creates a new latency metric
+func (QueueMetricProvider) NewLatencyMetric(string) workqueue.HistogramMetric {
+	return histgramWrapper{telemetry.NewHistogramWithOpts(
+		subsystem,
+		"queue_latency_seconds",
```
Comment (clamoriniere) [inline pkg/sbom/telemetry/telemetry.go:158]: I'm not sure we should add the unit in the metric name,

File: pkg/sbom/scanner/scanner.go (hunk 2 above)
Comment (AliDatadog) [reply pkg/sbom/scanner/scanner.go]: Renamed it accordingly 👍

File: pkg/sbom/telemetry/telemetry.go (hunk 3 above)
Comment (AliDatadog) [reply pkg/sbom/telemetry/telemetry.go]: Removed it

File: pkg/sbom/telemetry/telemetry.go [hunk 4]
```diff
@@ -82,3 +85,125 @@ var (
...
+	)}
+}
+
+// NewLatencyMetric creates a new latency metric
+func (QueueMetricProvider) NewLatencyMetric(string) workqueue.HistogramMetric {
+	return histgramWrapper{telemetry.NewHistogramWithOpts(
+		subsystem,
+		"queue_latency",
+		[]string{},
+		"SBOM queue latency",
```
Comment (clamoriniere) [inline pkg/sbom/telemetry/telemetry.go:160]: you can still add the unit in the description

Comment (AliDatadog) [reply pkg/sbom/telemetry/telemetry.go]: Done

**Review by clamoriniere — APPROVED** [review]

**Review by 0intro — APPROVED** [review]

**Pull request merged**